tools/build-release-tarball

```diff
@@ -62,6 +84,18 @@ git checkout-index -f -a --prefix "$TMPDIR/$prefix/"
 
 # Add the Git version information file
 ./tools/cache-zulip-git-version
+if [ -n "$TAG" ]; then
+    generated_version=$(head -n1 zulip-git-version)
+    base_version=$(head -n2 zulip-git-version | tail -n1)
+    if [ "$generated_version" != "$version" ]; then
+        echo "Version mismatch in generated version -- expected $version, got $generated_version!"
+        exit 1
+    fi
+    if [ "$base_version" != "$version" ]; then
+        echo "Version mismatch in generated version -- expected base version to be $base_version, got $base_version!"
```

Comment:
`got $base_version` → `got $version`.

This will always fail with our current procedure, since the commit hasn’t yet been pushed upstream.

Reply:
`expected base version to be $base_version` → `expected base version to be $version`, but you're right this will fail -- my testing I was doing wasn't with any local commits.

I think our options are:
1. Remove this check -- all of our previous releases have had a second line which is a previous commit, and have claimed in the version to be a fork from that version ~1 commit prior.  We change the logic for `is_fork` to say it isn't a fork if it's a `\d+\.\d+` tag.
2. Change `cache-zulip-git-version` to write the second line as the tag if the git describe was a tag, effectively moving the logic from (1) into the `zulip-git-version` file.
3. Change the release ordering to tag, push, build tarball, upload tarball -- which makes the `--tag` argument this PR adds impossible, unless `--tag` with a release-ish number also does a push.  Which feels wrong, in that that's not "building a release tarball" anymore, that's "doing a release."  So instead of this PR, we make a more general "release" tool, which does the equivalent of `git tag $version && git push && build-release-tarball $version && upload-release` to automate all of those together.

(3) seems like the best bet.

Reply:
Yeah, seems reasonable. A potential downside is that a `build-release-tarball` failure would burn a version number, but I trust our CI to make that unlikely.

Reply:
I guess we could get away with pushing just the commit up front, not the tag.

Reply:
Repushed with a `tools/release` to do this.